### PR TITLE
Stp/interface

### LINKED ISSUE #1000: Interface Improvements
- [x] Add information on which plays are valid, and their scores. Possibly also which invariants are considered to be true
- [x] Tab: all plays with all invariants (and their value) and all scores
- [x] Tab: all invariants and their values

### PR DESCRIPTION
**Pull request by Yuhanun**
Self-explanatory, updated the UI according to our new changes :)

Closes #1000

### CODE REVIEW

**Comment by RobotJente** [comment]
I got a segfault with this error after running your branch and clicking the invariants tab: 
Screenshot from 2020-04-30 17-05-45
Stack trace:
Screenshot from 2020-04-30 17-08-02

File: src/interface/widgets/PlaysWidget.cpp [hunk 1]
```diff
@@ -11,9 +11,12 @@ namespace rtt::ai::interface {
     }
 
     void PlaysWidget::updatePlays() {
-        std::lock_guard mtxLck{ dataMtx };
```
Comment (RobotJente) [inline src/interface/widgets/PlaysWidget.cpp]: Why did you remove the mutex?

File: src/interface/widgets/mainWindow.cpp [hunk 2]
```diff
@@ -180,9 +182,9 @@ void MainWindow::updateRobotsWidget() {
 
 void MainWindow::updatePlay(stp::Play *play) {
     stpWidget->updateContents(play);
+//    playsWidget->updatePlays();
+//    invariantsWidget->updateInvariants();
```
Comment (RobotJente) [inline src/interface/widgets/mainWindow.cpp]:
```suggestion
```

**Comment by RobotJente** [comment]
Branch is presumably failing because you committed an unresolved merge conflict xD

File: src/interface/widgets/PlaysWidget.cpp (hunk 1 above)
Comment (Yuhanun) [reply src/interface/widgets/PlaysWidget.cpp]: I no longer modify an internal string / buffer, I just write to the textbox directly

**Comment by Yuhanun** [comment]
Resolved.

**Review by timovdk — APPROVED** [review]

**Review by timovdk — APPROVED** [review]

**Comment by RobotJente** [comment]
Screenshot from 2020-05-19 10-36-48
The plays tab does not segfault anymore :+1:, however, it is also currently not displaying any useful information :stuck_out_tongue_closed_eyes:. It should display the play name, score, and if possible also their invariants?

**Review by timovdk — DISMISSED** [review]

File: include/roboteam_ai/interface/widgets/PlaysWidget.hpp [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+#ifndef RTT_PLAYSWIDGET_HPP
+#define RTT_PLAYSWIDGET_HPP
+
+#include <QtWidgets/QVBoxLayout>
+#include <QtWidgets/QTextEdit>
+
+namespace rtt::ai::interface {
+
+    class PlaysWidget : public QTextEdit {
```
Comment (HaicoDorenbos) [inline include/roboteam_ai/interface/widgets/PlaysWidget.hpp:13]: Where is the class documentation?

Comment (Yuhanun) [reply include/roboteam_ai/interface/widgets/PlaysWidget.hpp:13]: it's a temporary class i didn't put too much effort into it.

Comment (Yuhanun) [reply include/roboteam_ai/interface/widgets/PlaysWidget.hpp:13]: Plus it's pretty self-explanatory

**Review by timovdk — APPROVED** [review]

**Comment by rolfvdhulst** [comment]
Why is help needed for this?

**Review by rolfvdhulst — APPROVED** [review]
LGTM. Only wondering why in a previous PR getName() was determined to be char *. Why not use std::string or QString?

**Comment by timovdk** [comment]
It segfaults in the invariants, probably due to some sort of data race for world/robots/field. I think most segfaults stem from robots that don't exist anymore

**Comment by Yuhanun** [comment]
> LGTM. Only wondering why in a previous PR getName() was determined to be char *. Why not use std::string or QString?

I don't want O(n^2) allocations every tick.
